### PR TITLE
Io 4782 add mail api docs

### PR DESCRIPTION
**Pull request by GimmeYerGold**
CHANGES

• Added all of the Mail Endpoints I could find! (a lot!)

https://ioffice.myjetbrains.com/youtrack/issue/io-4782

### CODE REVIEW

File: api/endpoints/mail.md [hunk 1]
```diff
@@ -0,0 +1,898 @@
...
++ Response 200
+    [Mail][]    
+    
+## Add a Comment to a Mail Item [/mail/items///{id}///addComment]
+To add a Comment to a Mail Item send JSON with the required attributes.
+
+### Add a Comment to a Mail Item [PUT]
++ Parameters
+    + comment (string, `This is a comment!`) ... new comment for the Mail Item 
+    + commentBy (string, `Hank`) ... author noteg to accompany new comment for the Mail Item
```
Comment (quesojean) [inline api/endpoints/mail.md]: noteg correct spelling

Comment (quesojean) [inline api/endpoints/mail.md]: commentBy is not required.  If it is missing will use either the logged in user and if no authenticated user, will simply use "3rd Party"

File: api/endpoints/mail.md [hunk 2]
```diff
@@ -0,0 +1,898 @@
...
+            ]
+
+### Retrieve Mail Routes [GET]
+
++ Response 200
+    
+    [Mail Routes][] 
+    
+## Mail Locations [/mail/locations]
+Collection of all Mail Routes.
```
Comment (quesojean) [inline api/endpoints/mail.md]: Collection of all Mail Locations

File: api/endpoints/mail.md [hunk 3]
```diff
@@ -0,0 +1,898 @@
...
+
+            {
+              "comment: "This is a comment!",
+              "commentBy: "Bobby"
+            }
+
++ Response 200
+    [Mail][]                        
+        
+## Mail Items Collection [/mail/items]
```
Comment (quesojean) [inline api/endpoints/mail.md]: not sure if you want a separate entry or add optional request parameter to this endpoint, but we need to document searching mail endpoint (hits both mail and archived mail items):
will search id, trackingNumber, & originalTrackingNumber fields
mail/items?search=<text>

File: api/endpoints/mail.md [hunk 4]
```diff
@@ -0,0 +1,898 @@
...
+            }
+        } 
+        
++ Response 200     
+   [Mail][]        
+        
+## Deliver a Mail Item [/mail/items///{id}///deliver]
+To deliver a Mail Item send JSON with the required attributes.
+
+### Deliver a Mail Item [PUT]
```
Comment (quesojean) [inline api/endpoints/mail.md]: shouldnt we document optional properties as well?
SignatureData

File: api/endpoints/mail.md [hunk 5]
```diff
@@ -0,0 +1,898 @@
...
++ Parameters
```
Comment (quesojean) [inline api/endpoints/mail.md]: also, need to include batch deliver endpoint (PUT takes a Seq of DeliverData- see MailItemResource)

File: api/endpoints/mail.md [hunk 6]
```diff
@@ -0,0 +1,898 @@
...
+### List of all Mail Items [GET]
+
++ Response 200
+    
+    [Mail Items Collection][]
+    
+## Check In a Mail Item [/mail/items///{id}///checkin]
+To check in (create) a Mail Item send JSON with the required attributes.
+
+### Check In a Mail Item [POST]
```
Comment (quesojean) [inline api/endpoints/mail.md]: there is also a batch checkin post endpoint that takes a collection of mail items.
mail/items/checkin

I believe to checkin a single mail item, its simply a POST /mail/item with the mail json body

File: api/endpoints/mail.md [hunk 7]
```diff
@@ -0,0 +1,898 @@
...
+              "trackingNumber": "351987984"
+            }
+
++ Response 200
+    [Mail][]
+    
+## Pend a Mail Item [/mail/items///{id}///pending]
+To pend a Mail Item send JSON with the required attributes.
+
+### Pend a Mail Item [PUT]
```
Comment (quesojean) [inline api/endpoints/mail.md]: please document optional property as well pendingDate

File: api/endpoints/mail.md [hunk 8]
```diff
@@ -0,0 +1,898 @@
...
+            }
+
++ Response 200  
+    [Mail][] 
+    
+## Checkout a Mail Item [/mail/items///{id}///checkout]
+To checkout a Mail Item send JSON with the required attributes.
+
+### Checkout a Mail Item [PUT]
++ Parameters
```
Comment (quesojean) [inline api/endpoints/mail.md]: document optional  checkoutDate property

**Review by quesojean — CHANGES_REQUESTED** [review]
this is what i have so far...looks great, thank you!

File: api/endpoints/mail.md (hunk 4 above)
Comment (GimmeYerGold) [reply api/endpoints/mail.md]: 🎏 passing on optional properties for this PR 🎏

File: api/endpoints/mail.md (hunk 8 above)
Comment (GimmeYerGold) [reply api/endpoints/mail.md]: 🎏 passing on optional properties for this PR 🎏

File: api/endpoints/mail.md (hunk 7 above)
Comment (GimmeYerGold) [reply api/endpoints/mail.md]: 🎏 passing on optional properties for this PR 🎏

File: api/endpoints/mail.md (hunk 1 above)
Comment (GimmeYerGold) [reply api/endpoints/mail.md]: ✅

Comment (GimmeYerGold) [reply api/endpoints/mail.md]: ✅

File: api/endpoints/mail.md (hunk 2 above)
Comment (GimmeYerGold) [reply api/endpoints/mail.md]: ✅

File: api/endpoints/mail.md (hunk 3 above)
Comment (GimmeYerGold) [reply api/endpoints/mail.md]: ✅

File: api/endpoints/mail.md (hunk 6 above)
Comment (GimmeYerGold) [reply api/endpoints/mail.md]: ✅

File: api/endpoints/mail.md (hunk 5 above)
Comment (GimmeYerGold) [reply api/endpoints/mail.md]: ✅

**Review by quesojean — APPROVED** [review]

**Pull request merged**